enterprise/server/remote_execution/blockio/blockio.go

```diff
@@ -363,9 +364,25 @@ func (c *COWStore) ReadAt(p []byte, off int64) (int, error) {
 				p[i] = 0
 			}
 		} else {
-			if _, err := readFullAt(chunk, p[:readSize], chunkRelativeOffset); err != nil {
+			s, err := chunk.SizeBytes()
+			if err != nil {
 				return n, err
 			}
+			// Chunk might have less data available than the calculated size
+			// if this was the last chunk when we resized. If so then fill
+			// the range from [dataSize, readSize) with 0s.
+			dataSize := int64(readSize)
+			if remainder := s - chunkRelativeOffset; dataSize > remainder {
+				dataSize = max(0, remainder)
+			}
+			if dataSize > 0 {
+				if _, err := readFullAt(chunk, p[:dataSize], chunkRelativeOffset); err != nil {
+					return n, err
+				}
+			}
+			for off := dataSize; off < int64(readSize); off++ {
+				p[off] = 0
+			}
```

Comment:
@maggie-lou I know we talked a little bit about always making things chunk-aligned so that we wouldn't have to handle the last-chunk case here like this, but as I started to do the chunk alignment I started to second-guess whether that approach would cause more problems than it solved. e.g. it breaks a reasonable assumption that `size(file) == size(fileConvertedToCOW)` which could cause subtle bugs later.

That said, if this lazy approach seems too complex or seems like it will make the caching stuff more difficult, we can rethink this